src/utils.cpp

```diff
@@ -151,19 +151,27 @@ bool get_uint64_env_var(const std::string &str, uint64_t &dest)
 
 std::string get_pid_exe(pid_t pid)
 {
-  char proc_path[512];
-  char exe_path[4096];
-  int res;
+  std_filesystem::path proc_path{ "/proc" };
+  proc_path /= std::to_string(pid);
+  proc_path /= "exe";
 
-  sprintf(proc_path, "/proc/%d/exe", pid);
-  res = readlink(proc_path, exe_path, sizeof(exe_path));
-  if (res == -1)
+  try
+  {
+    if (!std_filesystem::exists(proc_path))
+      return "";
+
+    std::string exe_path = std_filesystem::read_symlink(proc_path).string();
+    if (exe_path.length() >= 4096)
+    {
+      throw std::runtime_error(
+          "executable path exceeded maximum supported size of 4096 characters");
+    }
+    return exe_path;
+  }
+  catch (const std_filesystem::filesystem_error &)
```

Comment:
I'd prefer having a `catch runtime_errror &err { throw err }` here to make it clear that you want to propagete the original error. Or maybe use the non-throwing alternatives for these filesystem calls